src/com/janek/basics/CalcGold.java

```diff
@@ -0,0 +1,101 @@
+package com.janek.basics;
+
+import static com.janek.basics.math.MathUtil.*;
+import static java.lang.Double.parseDouble;
+import static java.lang.String.format;
+import static java.lang.System.exit;
+import static java.lang.System.out;
+
+public class CalcGold {
+
+    private static final String INPUT_PLUS_SYMBOL = "+";
+    private static final String INPUT_PLUS_STRING = "dodać";
+    private static final String INPUT_MINUS_SYMBOL = "-";
+    private static final String INPUT_MINUS_STRING = "odjąć";
+    private static final String INPUT_X_SYMBOL = "x";
+    private static final String INPUT_X_STRING = "razy";
+    private static final String INPUT_SLASH_SYMBOL = "/";
+    private static final String INPUT_SLASH_STRING = "podzielić-przez";
+
+    public static final int VALIDATION_EXIT_CODE = 1;
+
+    public static void main(String[] args) {
+        validateArgsNumber(args);
+        validateSymbol(args);
+        validateNumber(args);
+
+        double result = calculateResult(args);
+        printResult(args, result);
+    }
+
+    private static void printResult(String[] args, double result) {
+        for (String arg : args) {
+            out.print(format("%s ", arg));
+        }
+        out.println(format("= %f", result));
+    }
+
+    private static double calculateResult(String[] args) {
+        double result = doCalculation(parseDouble(args[0]), parseSymbol(args[1]), parseDouble(args[2]));
+
+        for (int i = 3; i < args.length; i = i + 2) {
+            result = doCalculation(result, parseSymbol(args[i]), parseDouble(args[i + 1]));
+        }
+        return result;
+    }
+
+    private static void validateNumber(String[] args) {
+        for (int i = 0; i < args.length; i = i + 2) {
+            String number = args[i];
+            try {
+                parseDouble(number);
+            } catch (NumberFormatException e) {
+                out.println(format("Wpisałeś nieprawidłową liczbę! (%s)", number));
+                exit(VALIDATION_EXIT_CODE);
+            }
```

Comment:
So this is good to have the method which can test if the **string** is **number** in your `MathUtil` library. Please create the method:
```
public static boolean isNumber(String text)
```
in your library. Then use it instead of `try-catch` clause